test/test_executor.py

```diff
@@ -224,8 +225,9 @@ def test_execute_jobs():
             event_reactor.get_queue().put.reset_mock()
             jobs['one'].returncode = 0
             extensions = get_executor_extensions()
-            extensions[110]['extension2'].execute = Mock(return_value=0)
-            rc = execute_jobs(context, jobs)
+            extensions[110]['extension2'].execute = \
```

Comment:
I can't tell what the reason for `110` here is.  Is this just to ensure that you are above any other extensions that may have been in the list?

Reply:
Yes, the extensions returned by [get_executor_extensions](https://github.com/colcon/colcon-core/blob/35f04c4289c26b163f14989fe23989f6e7440a83/colcon_core/executor/__init__.py#L169-L177) are first grouped and ordered by their priorities and then in a dictionary with their name as the key. Some extension points do have a priority (to influence their order of use) while others don't (when there is no need for an order). For this specific test the extension has the priority 110 (see https://github.com/colcon/colcon-core/blob/cc235f558756c882b3d6f0076cb448c0f427dd3a/test/test_executor.py#L136) and therefore is returned in this nested dictionary under that priority group.